packages/framework/src/Command/DomainInfoCommand.php

```diff
@@ -0,0 +1,149 @@
+
+    protected const OPTION_ALL = 'all';
+
+    protected const RETURN_CODE_OK = 0;
+    protected const RETURN_CODE_ERROR = 1;
+
+    /**
+     * @var string
+     */
+    protected static $defaultName = 'shopsys:domains:info';
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Component\Domain\Domain
+     */
+    protected $domain;
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Component\Domain\Domain $domain
+     */
+    public function __construct(Domain $domain)
+    {
+        parent::__construct();
+
+        $this->domain = $domain;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure(): void
+    {
+        $this
+            ->setDescription('Loads and displays domain info.')
+            ->addArgument(static::ARG_PROPERTY_NAME, InputArgument::OPTIONAL, 'Property that should be loaded', 'id')
+            ->addArgument(static::ARG_ID, InputArgument::OPTIONAL, 'Domain ID (if omitted, the command will output all distinct values)')
+            ->addOption(static::OPTION_ALL, 'a', InputOption::VALUE_NONE, 'Display all property values (without sorting and deduplication)');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $io = new SymfonyStyle($input, $output);
+
+        try {
+            $domainConfigs = $this->getDomainConfigs($input);
+        } catch (\InvalidArgumentException $e) {
+            $io->error($e->getMessage());
+
+            return static::RETURN_CODE_ERROR;
+        }
+
+        $propertyAccessor = PropertyAccess::createPropertyAccessor();
+        $propertyName = $input->getArgument(static::ARG_PROPERTY_NAME);
+        $propertyValues = [];
+        foreach ($domainConfigs as $domainConfig) {
+            if (!$propertyAccessor->isReadable($domainConfig, $propertyName)) {
+                $this->outputPropertyNotAccessible($io, $domainConfig, $propertyName);
+
+                return static::RETURN_CODE_ERROR;
+            }
+
+            $propertyValues[] = $propertyAccessor->getValue($domainConfig, $propertyName);
+        }
+
+        if (!$input->getOption(static::OPTION_ALL)) {
+            sort($propertyValues);
+            $propertyValues = array_unique($propertyValues);
+        }
+        $io->writeln(implode("\t", $this->formatPropertyValues($propertyValues)));
+
+        return static::RETURN_CODE_OK;
+    }
+
+    /**
+     * @param \Symfony\Component\Console\Input\InputInterface $input
+     * @return \Shopsys\FrameworkBundle\Component\Domain\Config\DomainConfig[]
+     */
+    protected function getDomainConfigs(InputInterface $input): array
+    {
+        $domainConfigs = $this->domain->getAllIncludingDomainConfigsWithoutDataCreated();
+
+        $domainId = $input->getArgument(static::ARG_ID);
+        if ($domainId !== null) {
+            foreach ($domainConfigs as $domainConfig) {
+                if ($domainId === (string)$domainConfig->getId()) {
```

Comment:
```suggestion
                if ($domainConfig->getId() === (int)$domainId) {
```
(if you accept this suggestion, don't forget to change sprintf on line 111)

Reply:
I would rather make it fail when domain ID `"1not-really-a-number"` is provided

Reply:
that's fine by me :)